### PR TITLE
Rework ITextInputMethodClient

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

This PR adjusts ITextInputMethodClient so all platforms consume a unified API

- Add TextSelection Get, Set
- Remove CompositionRegion
- Remove ITextEditable
- Separate selection from surrounding text to be able to update them individually

## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036822-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs [hunk 1]
```diff
@@ -10,57 +8,81 @@ public interface ITextInputMethodClient
...
-        public int CursorOffset { get; set; }
-        public int AnchorOffset { get; set; }
+        /// <summary>
+        /// Inserts text at current cursor position.
+        /// </summary>
+        /// <param name="text">The inserted text</param>
+        /// <remarks>
+        /// Replaces currently selected text range.
+        /// </remarks>
+        void Insert(string? text);
```
Comment (kekekeks) [inline src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs:69]: This skips the routed events infrastructure, so it prevents people from intercepting the event. This is a breaking change and doesn't match UWP/WPF.

File: src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs [hunk 2]
```diff
@@ -10,57 +8,81 @@ public interface ITextInputMethodClient
...
+
+        /// <summary>
+        /// Gets or sets the composition region.
+        /// </summary>
+        /// <remarks>
+        /// This is used to highlight currently active composition region.
+        /// </remarks>
+        CompositionRegion CompositionRegion { get; set; }
```
Comment (kekekeks) [inline src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs:77]: Shouldn't the composition be controlled by the input method?

File: src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs (hunk 1 above)
Comment (Gillibald) [reply src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs:69]: We can raise the event on TopLevel. We should still unify this implementation.

File: src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs (hunk 2 above)
Comment (Gillibald) [reply src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs:77]: This state is shared by the platform IME and the TextPresenter. The region should only be set by the IME. This is mainly used to highlight the currently active composition region. We can split this into `CompositionRegion { get; }` `SetCompositionRegion(CompositionRegion)` Like PreeditText

Comment (Gillibald) [reply src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs:77]: I have changed it to just expose SetCompositionRegion(CompositionRegion)

File: src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs (hunk 1 above)
Comment (Gillibald) [reply src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs]: Removed

File: src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs (hunk 2 above)
Comment (Gillibald) [reply src/Avalonia.Base/Input/TextInput/ITextInputMethodClient.cs]: Removed

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036926-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036936-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036943-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036947-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036981-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036983-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0036997-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037007-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037089-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037119-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037137-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037185-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037219-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0037246-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Pull request merged**